query/sources/common/src/main/java/org/codice/ddf/admin/sources/utils/RequestUtils.java

```diff
@@ -122,24 +120,22 @@
      * @param webClient {@code WebClient} to send POST request with
      * @param urlField  original request url field
      * @param content   Body of the post request
-     * @return a {@link ReportWithResultImpl} containing a {@link ResponseField} or an {@link org.codice.ddf.admin.api.report.ErrorMessage} on failure.
+     * @return a {@link Report} containing a {@link ResponseField} or an {@link org.codice.ddf.admin.api.report.ErrorMessage} on failure.
      */
-    public ReportWithResultImpl<ResponseField> sendPostRequest(WebClient webClient,
+    public Report<ResponseField> sendPostRequest(WebClient webClient,
             UrlField urlField, String content) {
-        ReportWithResultImpl<ResponseField> responseResult = new ReportWithResultImpl<>();
-        responseResult.addMessages(endpointIsReachable(urlField));
-        if (responseResult.containsErrorMsgs()) {
+        Report endpointIsReachableReport = endpointIsReachable(urlField);
+        Report<ResponseField> responseResult = Reports.fromErrors(endpointIsReachableReport);
+        if (responseResult.containsErrorMessages()) {
```

Comment:
Since we cleaned up the return statements below, Can't we check the `endpointIsReachableReport.containsErrorMessages()` here and then only make a responseResult on line 130 from the `endpointIsReachableReport` if we need to? ie, `return Reports.fromErrors(endpointIsReachableReport)` on line 130 instead of making it on line 128.